packages/components/src/navigation/index.js

```diff
@@ -121,6 +121,96 @@ const Navigation = ( { activeItemId, children, data, rootTitle } ) => {
 	);
 };
 
+export function Navigation2( { children } ) {
+	const [ activeLevel, setActiveLevel ] = useState( 'root' );
+	const [ slideOrigin, setSlideOrigin ] = useState( 'left' );
+
+	const isMounted = useRef( false );
+	useEffect( () => {
+		if ( ! isMounted.current ) {
+			isMounted.current = true;
+		}
+	}, [] );
+
+	const navigateTo = ( level ) => {
+		setActiveLevel( level );
+		setSlideOrigin( 'left' );
+	};
+
+	const navigateBack = ( level ) => {
+		setActiveLevel( level );
+		setSlideOrigin( 'right' );
+	};
+
+	const NavigationBackButton = ( { parentLevel, label } ) => {
+		return (
+			<Button
+				className="components-navigation__back-button"
+				isPrimary
+				onClick={ () => navigateBack( parentLevel ) }
+			>
+				{ label }
+			</Button>
+		);
+	};
+
+	return (
+		<Root className="components-navigation">
+			<Animate
+				key={ activeLevel }
+				type="slide-in"
+				options={ { origin: slideOrigin } }
+			>
+				{ ( { className: animateClassName } ) => (
+					<div
+						className={ classnames( {
+							[ animateClassName ]: isMounted.current,
+						} ) }
+					>
+						{ children( {
+							activeLevel,
+							navigateTo,
+							NavigationBackButton,
+						} ) }
+					</div>
+				) }
+			</Animate>
+		</Root>
+	);
+}
+
+export function Navigation2Level( {
+	children,
+	slug,
+	title,
+	activeLevel,
+	navigateBack,
+} ) {
+	if ( activeLevel !== slug ) {
+		return null;
+	}
+
+	return (
+		<div className="components-navigation__level">
+			{ navigateBack }
+			<h1>{ title }</h1>
+			<MenuUI>{ children }</MenuUI>
+		</div>
+	);
+}
+
+export function Navigation2Item( { slug, title, onClick, activeItem } ) {
+	const classes = classnames( 'components-navigation__menu-item', {
+		'is-active': activeItem === slug,
+	} );
+
+	return (
+		<MenuItemUI className={ classes }>
+			<Button onClick={ onClick }>{ title }</Button>
+		</MenuItemUI>
+	);
+}
+
```

Comment:
Just thinking out loud here:

There are two kinds of navigation items at play: One that actually navigates and one that changes level. The first has no children and has `setActiveItem` in its onClick. The second has children, has `navigateTo` in its onClick and will render a right chevron. What if we separated them into distinct components?

```js
<Navigation2Item
	slug="item-3"
	title="Item 3"
	onClick={ () => {
		setActiveItem( 'item-3' );
	} }
	activeItem={ activeItem }
/>
<Navigation2GroupItem
	slug="item-4"
	title="Item 4"
	navigateTo="nested-sub-menu"
/>
```
This allows `Navigation2GroupItem` to always render a chevron and have a clear set of props. It could be passed down in the render function so `navigateTo` can be part of its button's onClick.

Reply:
Love this!
I might be inclined to have just one component with an additional prop indicating parent navigation, to avoid exposing too many components which I think could be cumbersome to learn and remember.

Reply:
I also like this idea. It would be nice to expose the `NavigationBackButton` component for consumption leaving it with an `onClick` prop so we can resolve the back button not implemented in https://github.com/woocommerce/navigation/pull/68.